File: marks/models/dbobj/assessments.py
from datetime import datetime
from pydantic import BaseModel

class AssessmentsBDObj(BaseModel):
    id: int
    course_id: int
    name: str
    assessment_type_id: int
    max_marks: int
    is_marks_published: bool
    assessment_date: datetime
    created_by_id: int
    created_at: datetime
    updated_at: datetime

File: frontend/lib/types/assessments.ts
export interface CreateAssessmentRequest {
  name: string;
  assessment_type_id: number;
  max_marks: number;
  is_marks_published: boolean;
  assessment_date: Date;
}

export interface AssessmentDBObject {
  id: number;
  course_id: number;
  name: string;
  assessment_type_id: number;
  max_marks: number;
  is_marks_published: boolean;
  assessment_date: Date;
  created_by_id: number;
  created_at: Date;
  updated_at: Date;
}

export interface UpdateAssessmentRequest {
  name?: string;
  assessment_type_id?: number;
  max_marks?: number;
  is_marks_published?: boolean;
  assessment_date?: Date;
}
